sig-and-wg/wg-knative/knative-eventing-adoption.md

```diff
@@ -0,0 +1,180 @@
+
+
+# Provisioning a messaging solution
+The provisioning of various messaging solutions such as NATS Streaming, Google PubSub, Kafka etc. is abstracted from the production and consumption of events in Kyma.
+
+Messaging solutions can be provisioned at:
+
+* During installation
+* While Kyma is running
+
+The operator can provision messaging system using:
+
+* Helm Charts
+* Service Catalog
+* Plain K8S deployments
+
+In the long run, we can build utils that automate the provisioning process. For now we can start simple
+
+A typical provisioner looks like:
+
+![](assets/provisiner.svg)
+
+> **TBD** The deprovisioning flow. The impact on the existing event types needs to be understood first.
+
+# Kyma event types
+Before Knative adoption, there was no need to create any metadata such as topics or channels in the Kyma event bus as the event types were mapped to NATS Streaming subjects while publishing an event.
+
+With Knative adoption, this model can no longer be applied:
+
+* Knative eventing requires to create Channels as a pre-step to publish. The channels are heavy objects with cascading resources. 
+* The model is Kyma needs to be generic and in-sync with Knative eventing to enable plugging in various messaging solutions.
+
+## Option 1
+* `1 Kyma event-type` is mapped to a `1 Knative channel`.
+
+**Pros**
+
+* Simple approach.
+* Aligned with Knative eventing concepts and underlying bus implementations.
+  ![](assets/mapping-refer.svg)
+* Some reference implementations such as `Kafka` are mapping one `Topic` to a `Knative Channel`
+* **This enables us to build a thin abstraction on top of knative**.
+  * Kyma does not want to run heavy workloads on the cluster when the customer is using a `Cloud PubSub`.
+* The Knative subscription object is mapped to channel where a single channel can have many subscriptions.
+  * Following this approach we can keep the subscription model simple.
+
+**Cons**
+
+* A Knative channel is a heavy object with many cascading resources. Having many channels can increase load on istio service mesh. 
+
+## Option 2
+
+* Map multiple `Kyma event-types` to a `single channel` or do some kind of grouping.
+
+**Pros**
+
+* Create minimal knative channels thus creating less cascading resources.
+
+**Cons**
+
+* The extra complexity of maintaining the mapping.
+* **The Kyma eventing will be a thick layer with potential heavy workloads** running on cluster despite customer using a `Cloud PubSub`.
+* The subscription management will become complex.
+  * A single knative channel will have subscriptions for multiple event-types.
+  * This will generate unnecessary network traffic and workload.
+  * The `dispatcher` will receive events for all event-types belonging to a channel. Then it has to discard and only deliver those for which the Kyma Subscription has been created.
+  * This will lead to fat topics in the underlying PubSub.
```

Comment:
**Comment**: what are `fat topics`? Also I would combine this sentence with the previous one.